src/libraries/System.Private.CoreLib/src/System/Numerics/Matrix4x4.cs

```diff
@@ -1803,188 +1803,270 @@ public static bool Decompose(Matrix4x4 matrix, out Vector3 scale, out Quaternion
         {
             bool result = true;
 
-            unsafe
+            float det;
```

Comment:
It would be interesting to compare this to `XMMatrixDecompose` from DirectX Math: https://github.com/microsoft/DirectXMath/blob/main/Inc/DirectXMathMatrix.inl#L1111-L1206